### ISSUE TITLE
NuGetPack fails if no Files have been specified

### ISSUE THREAD

**Issue by wieslawsoltes** [cake-build/cake#1154]
```C#
var nuspec = new NuGetPackSettings()
{
    Id = "SomePackage",
    Version = "0.1.0",

    Authors = new [] { "a" },
    Owners = new [] { "a" },
    LicenseUrl = new Uri("http://opensource.org/licenses/MIT"),
    ProjectUrl = new Uri("https://github.com/a/a/"),
    RequireLicenseAcceptance = false,
    Symbols = false,
    NoPackageAnalysis = true,
    Description = "a",
    Copyright = "Copyright 2015",
    Tags = new [] { "a" },
    Dependencies = new []
    {
        new NuSpecDependency() { Id = "Dep1", Version = "0.1.0" },
        new NuSpecDependency() { Id = "Dep2", Version = "0.1.0" }
    },
    OutputDirectory = (DirectoryPath)Directory("./artifacts")
};
NuGetPack(nuspec);
```

```
Error: 
Required setting Files not specified
```

This should be allowed, very useful for creating meta packages.

**Comment by gep13** [comment]
@wieslawsoltes agreed.  This is definitely an issue.  If there are dependencies defined, then Files is not a required parameter.

**Comment by devlead** [comment]
Fixed by #1154
